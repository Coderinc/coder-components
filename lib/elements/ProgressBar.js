import React from 'react';
import styled from 'styled-components';

const ProgressWrapper = styled.div`
  background-color: ${props => props.theme.offsetBackground};

  height: 0.75rem;
  width: 100%;

  border-radius: 0.375rem;
`;

const Progress = styled.div`
  background-color: ${props => props.theme.primary};

  height: 100%;
  width: ${props => props.percent || 0}%;

  border-radius: 0.5rem;

  transition: width 0.25s linear;
`;

export const ProgressBar = props => (
  <ProgressWrapper>
    <Progress {...props} />
  </ProgressWrapper>
);

export default ProgressBar;
