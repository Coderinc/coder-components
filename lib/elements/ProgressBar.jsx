import React from 'react';
import { Motion, spring } from 'react-motion';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ProgressWrapper = styled.div`
  background-color: ${({ theme }) => theme.offsetBackground};

  height: 0.75rem;
  width: 100%;

  border-radius: 0.375rem;
`;

const Progress = styled.div`
  background-color: ${({ theme }) => theme.primary};

  height: 100%;

  border-radius: 0.5rem;
`;

class ProgressBar extends React.Component {
  render() {
    const { percent } = this.props;

    return (
      <ProgressWrapper {...this.props}>
        <Motion style={{ percent: spring(percent) }}>
          {({ percent }) => <Progress style={{ width: `${percent}%` }} />}
        </Motion>
      </ProgressWrapper>
    );
  }
}

// const ProgressBar = ({ percent, ...props }) => (
//   <ProgressWrapper {...props}>
//     <Motion defaultStyle={{ >
//       <Progress percent={percent} />
//     </Motion>
//   </ProgressWrapper>
// );

ProgressBar.propTypes = {
  percent: PropTypes.number,
};

ProgressBar.defaultProps = {
  percent: 0,
};

export default ProgressBar;
