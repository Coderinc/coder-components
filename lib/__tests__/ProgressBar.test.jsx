import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import ProgressBar from '../elements/ProgressBar';

describe('ProgressBar', () => {
  it('renders correctly', () => {
    const component = renderer.create(<ProgressBar percent={50} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
