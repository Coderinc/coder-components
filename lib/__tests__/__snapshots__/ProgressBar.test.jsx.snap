// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders correctly 1`] = `
.c0 {
  height: 0.75rem;
  width: 100%;
  border-radius: 0.375rem;
}

.c1 {
  height: 100%;
  width: 50%;
  border-radius: 0.5rem;
  -webkit-transition: width 0.25s linear;
  transition: width 0.25s linear;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;
